numpy/distutils/system_info.py

```diff
@@ -498,17 +505,46 @@ def calc_libraries_info(self):
                 dict_append(info, **i)
             else:
                 log.info('Library %s was not found. Ignoring' % (lib))
+            i = self.check_libs(r_dirs, [lib])
+            if i is not None:
+                # Swap library keywords found to runtime_library_dirs
+                # the libraries are insisting on the user having defined
+                # them using the library_dirs, and not necessarily by
+                # runtime_library_dirs
+                del i['libraries']
+                i['runtime_library_dirs'] = i.pop('library_dirs')
+                dict_append(info, **i)
```

Comment:
It was more to be consistent in that if later revisions of check_libs
returns more than the two keys (libraries and library_dirs). In that
_future_ case would any additional keys also get passed. Hence the obscure
delete, pop sequence.
It may be overkill?

2015-02-25 9:53 GMT+01:00 Ralf Gommers <notifications@github.com>:

> In numpy/distutils/system_info.py
> <https://github.com/numpy/numpy/pull/5597#discussion_r25325312>:
>
> > @@ -498,17 +505,46 @@ def calc_libraries_info(self):
> >                  dict_append(info, **i)
> >              else:
> >                  log.info('Library %s was not found. Ignoring' % (lib))
> > +            i = self.check_libs(r_dirs, [lib])
> > +            if i is not None:
> > +                # Swap library keywords found to runtime_library_dirs
> > +                # the libraries are insisting on the user having defined
> > +                # them using the library_dirs, and not necessarily by
> > +                # runtime_library_dirs
> > +                del i['libraries']
> > +                i['runtime_library_dirs'] = i.pop('library_dirs')
> > +                dict_append(info, **i)
>
> The above looks correct, but it took me a couple of minutes to understand
> the comment and the use of del. How about replacing the whole if clause
> with:
>
> # From `check_libs` we get keys "libraries" and "library_dirs",
> # while we need only key "runtime_library_dirs" with those paths
> dict_append(info, runtime_library_dirs=i['library_dirs'])
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/5597/files#r25325312>.
>



-- 
Kind regards Nick